modules/@apostrophecms/schema/index.js

```diff
@@ -611,92 +616,207 @@ module.exports = {
+            ? error.path.split('.')
+            : [ null, error.path ];
+
+          const curDestination = destId
+            ? destination.find(({ _id }) => _id === destId)
+            : destination;
+
+          const errorPath = destinationPath
+            ? `${destinationPath}.${error.path}`
+            : error.path;
+
+          // Case were this error field hasn't been treated
+          // Should check if path starts with, because parent can be invisible
+          const nonVisibleField = hiddenAncestors || nonVisibleFields.has(errorPath);
+
+          // We set default values only on final error fields
+          if (nonVisibleField && !error.data?.errors) {
+            const curSchema = self.getFieldLevelSchema(schema, error.schemaPath);
+            self.setDefaultToInvisibleField(curDestination, curSchema, error.path);
+            continue;
+          }
+
+          if (error.data?.errors) {
+            const subErrors = await self.handleConvertErrors({
+              req,
+              schema,
+              convertErrors: error.data.errors,
+              nonVisibleFields,
+              destination: curDestination[destPath],
+              destinationPath: errorPath,
+              hiddenAncestors: nonVisibleField
+            });
+
+            // If invalid error has no sub error, this one can be removed
+            if (!subErrors.length) {
               continue;
             }
-          }
 
-          if (!Array.isArray(error) && typeof error !== 'string') {
-            self.apos.util.error(error + '\n\n' + error.stack);
+            error.data.errors = subErrors;
           }
-          errorsList.push(error);
+          validErrors.push(error);
+        }
+
+        return validErrors;
+      },
+
+      setDefaultToInvisibleField(destination, schema, fieldPath) {
+        // Field path might contain the ID of the object in which it is contained
+        // We just want the field name here
+        const [ _id, fieldName ] = fieldPath.includes('.')
+          ? fieldPath.split('.')
+          : [ null, fieldPath ];
+        // It is not reasonable to enforce required,
+        // min, max or anything else for fields
+        // hidden via "if" as the user cannot correct it
+        // and it will not be used. If the user changes
+        // the conditional field later then they won't
+        // be able to save until the erroneous field
+        // is corrected
+        const field = schema.find(field => field.name === fieldName);
+        if (field) {
+          // To protect against security issues, an invalid value
+          // for a field that is not visible should be quietly discarded.
+          // We only worry about this if the value is not valid, as otherwise
+          // it's a kindness to save the work so the user can toggle back to it
+          destination[field.name] = klona((field.def !== undefined)
+            ? field.def
+            : self.fieldTypes[field.type]?.def);
```

Comment:
As descibed above, not sure where this method is useful except for strings have a `pattern` property.